internal/topology/variables/cluster_variable_validation.go

```diff
@@ -92,24 +94,24 @@ func validateClusterVariable(clusterVariable *clusterv1.ClusterVariable, cluster
 	// Parse JSON value.
 	var variableValue interface{}
 	if err := json.Unmarshal(clusterVariable.Value.Raw, &variableValue); err != nil {
-		return field.ErrorList{field.Invalid(fldPath, string(clusterVariable.Value.Raw),
-			fmt.Sprintf("variable %q could not be parsed: %v", clusterVariable.Name, err))}
+		return field.ErrorList{field.Invalid(fldPath.Child("value"), string(clusterVariable.Value.Raw),
+			fmt.Sprintf("value for variable %q could not be parsed: %v", clusterVariable.Name, err))}
 	}
 
 	// Convert schema to Kubernetes APIExtensions Schema.
 	apiExtensionsSchema, err := convertToAPIExtensionsJSONSchemaProps(&clusterClassVariable.Schema.OpenAPIV3Schema)
 	if err != nil {
-		return field.ErrorList{field.Invalid(fldPath, string(clusterVariable.Value.Raw),
-			fmt.Sprintf("invalid schema in ClusterClass for variable %q: failed to convert schema %v", clusterVariable.Name, err))}
+		return field.ErrorList{field.Invalid(fldPath, clusterVariable,
```

Comment:
This leads now to the following error text:

> spec.topology.variables[0]: Invalid value: v1beta1.ClusterVariable{Name:"cpu", Value:v1.JSON{Raw:[]uint8{0x31}}}: failed to convert schema definition for variable "cpu"; ClusterClass should be checked: <nil>,

I wonder if we should use `field.InternalError` instead

```go
return field.ErrorList{field.InternalError(fldPath,
			errors.Wrapf(err, "failed to convert schema definition for variable %q; ClusterClass should be checked", clusterVariable.Name))} // TODO: consider if to add ClusterClass name
```

> spec.topology.variables[0]: Internal error: failed to convert schema definition for variable "cpu"; ClusterClass should be checked: <conversion error text>,